### PR TITLE
feat: remove unused code and update query string on tab change

### PR DESCRIPTION
**Pull request by pxlrbt**

### CODE REVIEW

File: components/Layout.js [hunk 1]
```diff
@@ -10,45 +10,38 @@ import React, { useState, useEffect, useRef } from 'react'
...
-  };
 
-  useEffect(() => {
-    if (typeof window === 'undefined') {
-      return
-    }
+    const params = new URLSearchParams(location.search);
+    params.set('frontend', value.frontend);
+    params.set('backend', value.backend);
+    history.replaceState(null, null, '?' + params.toString());
```
Comment (RobertBoes) [inline components/Layout.js:33]: Unfortunately this causes the forward/back navigation to stop working. I can use history navigation, the URL changes, but the page remains the same

**Review by RobertBoes — CHANGES_REQUESTED** [review]
I tried figuring out why the history state doesn't work properly, but it seems like you'd need some sort of state object instead of null

**Comment by pxlrbt** [comment]
The state can be null. Probably it's the url. I only provide the search string instead of the full url.

**Comment by RobertBoes** [comment]
But the query is appended properly, it's not overwriting the complete URL so that's how it should work.. Can't seem to figure out why it bugs out the history tho

**Comment by pxlrbt** [comment]
You're right. It was the history state. Apparently NextJS is using it internally. I fixed it by passing on the current state.

**Review by pxlrbt — COMMENTED** [review]
Fixed.

**Review by RobertBoes — APPROVED** [review]

**Pull request merged**